app/scripts/lib/typed-message-manager.js

```diff
@@ -132,14 +129,33 @@ module.exports = class TypedMessageManager extends EventEmitter {
    *
    */
   validateParams (params) {
-    assert.equal(typeof params, 'object', 'Params should ben an object.')
-    assert.ok('data' in params, 'Params must include a data field.')
-    assert.ok('from' in params, 'Params must include a from field.')
-    assert.ok(Array.isArray(params.data), 'Data should be an array.')
-    assert.equal(typeof params.from, 'string', 'From field must be a string.')
-    assert.doesNotThrow(() => {
-      sigUtil.typedSignatureHash(params.data)
-    }, 'Expected EIP712 typed data')
+    switch (params.version) {
```

Comment:
For now, since there's only two versions of the proposal, we include both validation logic paths in the same TypedMessageManager class. If there ends up being even more iterations of the proposal with different validation schemes (unlikely, but possible), we can break this class down into a base Manager class and multiple implementations that extend it with a custom `validateParams` method..